python/ray/data/_internal/logical/optimizers.py

```diff
@@ -0,0 +1,47 @@
+from typing import List
+
+from ray.data._internal.execution.interfaces import PhysicalOperator
+from ray.data._internal.logical.interfaces import Rule, Optimizer, LogicalOperator
+from ray.data._internal.logical.planner import Planner
+
+
+class LogicalOptimizer(Optimizer):
+    """The optimizer for logical operators."""
+
+    @property
+    def rules(self) -> List[Rule]:
+        # TODO: Add logical optimizer rules.
+        return []
+
+
+class PhysicalOptimizer(Optimizer):
+    """The optimizer for physical operators."""
+
+    @property
+    def rules(self) -> List["Rule"]:
+        # TODO: Add physical optimizer rules.
+        return []
+
+
+class LogicalPlan:
+    """The plan with a DAG of logical operators."""
+
+    def __init__(self, dag: LogicalOperator):
+        self._dag = dag
+
+    @property
+    def dag(self) -> LogicalOperator:
+        """Get the DAG of logical operators."""
+        return self._dag
+
+    def get_execution_dag(self) -> PhysicalOperator:
```

Comment:
This looks no need to be a class member and can be a function outside.

Reply:
@jianoaix - changed to a utility function.